build_modules/README.md

```diff
@@ -32,3 +32,70 @@ building your Dart code.
 
 [`build_web_compilers`]: https://pub.dartlang.org/packages/build_web_compilers
 [`build_vm_compilers`]: https://pub.dartlang.org/packages/build_vm_compilers
+
+## Module algorithms
+
+The overall process for emitting modules follows these steps:
+
+1. Emit a `.module.library` asset for every `.dart` library containing a
+   description of the directives (imports, exports, and parts), a list of the
+   `dart:` imports used, and a determination of whether the library is an
+   "entrypoint". Entrypoint libraries are either those which are likely to be
+   directly imported (they are under `lib/` but not `lib/src/`) or which have a
+   `main`. Only the libraries that can be imported (they aren't part files) will
+   have an output. This step is mainly separated out for better invalidation
+   behavior since the output will change less frequently than the code. The
+   outputs from this step are non differentiated by platform.
+1. Emit package level `.meta_module` assets. This is an aggregation of the
+   `.module.library` information with a first run at creating a module
+   structure. The output depends on the algorithm, described below. The outputs
+   from this step are specific to a platform and will exclude any libraries
```

Comment:
I don't think we exclude anything? We just mark them as not supported.

Reply:
Good catch. Fixed.